imap-listener/src/index.ts

```diff
@@ -0,0 +1,321 @@
+            continue
+          }
+
+          // Log message details to console (TEST output)
+          log.info("Processing email", {
+            uid: message.uid,
+            from: message.envelope?.from?.[0]?.address || "unknown",
+            subject: message.envelope?.subject || "(no subject)",
+            date: message.envelope?.date?.toISOString() || "unknown",
+            size: message.source?.length || 0,
+          })
+
+          // Mark as seen
+          await client.messageFlagsAdd(uid.toString(), ["\\Seen"])
+          log.info("Marked message as seen", { uid: message.uid })
+
+          // Move to DONE folder
+          await client.messageMove(uid.toString(), config.folders.done)
+          log.success("Moved message to DONE folder", { uid: message.uid })
+
+          // Delay between messages
+          if (config.processing.delay > 0) {
+            log.info("Waiting before processing next message", { delayMs: config.processing.delay })
+            await new Promise((resolve) => setTimeout(resolve, config.processing.delay))
+          }
+        } catch (error) {
+          log.error("Error processing individual message", error, { uid })
+        }
+      }
+    } finally {
+      lock.release()
+    }
+
+  } catch (error) {
+    log.error('Error processing unseen mails', error);
+  } finally {
+    isProcessing = false;
+  }
+}
+
+async function verifyImapFolders(client: ImapFlow): Promise<boolean> {
+  try {
+    log.info('Verifying IMAP folder structure...');
+
+    const mailboxes: ListResponse[] = await client.list()
+    const existingFolders = mailboxes.map((mb) => mb.path)
+
+    log.info('Available IMAP folders', { availableFolders: existingFolders });
+
+    const requiredFolders = [config.folders.inbox, config.folders.done, config.folders.error]
+
+    const missingFolders = requiredFolders.filter(folder => !existingFolders.includes(folder));
+
+    if (missingFolders.length > 0) {
+      log.error("Required IMAP folders are missing", new Error("Missing folders"), {
+        missingFolders,
+        requiredFolders,
+        existingFolders,
+      })
+      return false
+    }
+
+    log.success('All required IMAP folders exist');
+    return true;
+  } catch (error) {
+    log.error('Error verifying IMAP folders', error);
+    return false;
+  }
+}
+
+async function startImapListener(): Promise<void> {
+  try {
+    log.info("Starting IMAP Listener Service", {
+      host: config.imap.host,
+      port: config.imap.port,
+      secure: config.imap.secure,
+    })
+
+    if (!config.imap.auth.user || !config.imap.auth.pass) {
+      throw new Error("IMAP_USER and IMAP_PASSWORD environment variables are required")
+    }
+
+    const imapConfig: ImapFlowOptions = {
+      host: config.imap.host,
+      port: config.imap.port,
+      secure: config.imap.secure,
+      auth: {
+        user: config.imap.auth.user,
+        pass: config.imap.auth.pass,
+      },
+    }
+    client = new ImapFlow(imapConfig)
```

Comment:
```suggestion
    client = new ImapFlow(config.imap)
```

Der Vorteil der types oben is ja, dass wir uns hier darauf verlassen können, dass sie stimmen. Sehe keine Vorteil sie nochmal aufzulisten.